### PR TITLE
ASR EdgeZone to EdgeZone support

### PR DESCRIPTION
**Pull request by shesamian**
## Description

- Modify the enable protection to also accept `RecoveryExtendedLocation` as parameter
- Modify the create recovery plan to add three new scenarios:
   - EdgeZoneToEdgeZone
   - EdgeZoneToAzure
   - AzureToEdgeZone
- ~Removed `-RecoveryVmName` from A2A parameter set of ASR enable protection~

## Checklist

- [X] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [X] The title of the PR is clear and informative
- [X] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [X] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [X] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [X] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by NoriZC** [comment]
Hi, this draft PR has not been taken care of for long. Could you please check and close it? Thanks.

File: src/RecoveryServices/RecoveryServices/Az.RecoveryServices.psd1 [hunk 1]
```diff
@@ -12,7 +12,7 @@
 # RootModule = ''
 
 # Version number of this module.
-ModuleVersion = '5.4.1'
+ModuleVersion = '5.4.2'
```
Comment (isra-fel) [inline src/RecoveryServices/RecoveryServices/Az.RecoveryServices.psd1:15]: Versions are not meant to be updated manually. Please undo this.

File: src/RecoveryServices/RecoveryServices/ChangeLog.md [hunk 2]
```diff
@@ -24,6 +24,11 @@
 * Fixed output container in re-registration scenario for Register-AzRecoveryServicesBackupContainer cmdlet.
 * Added support for Enabling/Disabling the azure monitor and classic alerts for recovery services vault.
 
+## Version 5.4.2
```
Comment (isra-fel) [inline src/RecoveryServices/RecoveryServices/ChangeLog.md:27]: Please put your changelog under the "Upcoming Release" title, and do not add extra titles.

File: src/RecoveryServices/RecoveryServices.SiteRecovery/ReplicationProtectedItem/NewAzureRmRecoveryServicesAsrReplicationProtectedItem.cs [hunk 3]
```diff
@@ -179,8 +179,6 @@ public class NewAzureRmRecoveryServicesAsrReplicationProtectedItem : SiteRecover
         [Parameter(ParameterSetName = ASRParameterSets.HyperVSiteToAzure)]
         [Parameter(ParameterSetName = VMwareToAzureWithDiskType)]
         [Parameter(ParameterSetName = VMwareToAzureParameterSet)]
-        [Parameter(ParameterSetName = ASRParameterSets.AzureToAzure)]
```
Comment (isra-fel) [inline src/RecoveryServices/RecoveryServices.SiteRecovery/ReplicationProtectedItem/NewAzureRmRecoveryServicesAsrReplicationProtectedItem.cs:182]: Removing parameters (from parameter sets) is a breaking change and is not allowed outside of major releases. The next major release starts from Sep 6th, so please
1. Defer this PR until major release
2. Submit another PR to add [breaking change attributes ](https://github.com/Azure/azure-powershell/blob/main/documentation/breaking-changes/breaking-changes-attribute-help.md)that serve as preannouncement.

**Review by isra-fel — CHANGES_REQUESTED** [review]
First, make sure to [regenerate help](https://github.com/Azure/azure-powershell/blob/main/documentation/development-docs/help-generation.md) whenever you make any change to the cmdlet syntax

File: src/RecoveryServices/RecoveryServices.SiteRecovery/ReplicationProtectedItem/NewAzureRmRecoveryServicesAsrReplicationProtectedItem.cs (hunk 3 above)
Comment (shesamian) [reply src/RecoveryServices/RecoveryServices.SiteRecovery/ReplicationProtectedItem/NewAzureRmRecoveryServicesAsrReplicationProtectedItem.cs:182]: @isra-fel I am wondering if have you had a chance to check out my recent changes. Thank you so much.

**Comment by wyunchi-ms** [comment]
Please fix the failed test cases and static analysis

**Comment by viananth** [comment]
azp /run

**Comment by shesamian** [comment]
@wyunchi-ms can you please re-run the pipelines so we can see what is wrong with the BVTs?

**Comment by shesamian** [comment]
@wyunchi-ms the bot says "No pipelines are associated with this pull request." I am not sure.

**Comment by wyunchi-ms** [comment]
Please fix the failed test cases and static analysis. And retarget branch to release-2022-10-12 if you want to go into the release.

**Comment by wyunchi-ms** [comment]
The release branch is closed. Please fix the issues and wait for next release.

**Comment by shesamian** [comment]
@wyunchi-ms can you please checkout the PR again? I think the BVT tests and static checker are fixed

**Comment by VeryEarly** [comment]
azure-powershell - security-tools

**Review by VeryEarly — APPROVED** [review]

**Pull request merged**